pkg/crane/tag.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package crane
+
+import (
+	"fmt"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+)
+
+// Tag adds tag to the remote img.
+func Tag(img, tag string) error {
+	ref, err := name.ParseReference(img)
+	if err != nil {
+		return fmt.Errorf("parsing reference %q: %v", img, err)
+	}
+	desc, err := remote.Get(ref, remote.WithAuthFromKeychain(authn.DefaultKeychain))
+	if err != nil {
+		return fmt.Errorf("fetching %q: %v", img, err)
+	}
+
+	dst := ref.Context().Tag(tag)
+
+	return remote.Tag(dst, &wrap{desc}, remote.WithAuthFromKeychain(authn.DefaultKeychain))
+}
+
+// wrap coerces a remote.Descriptor into a remote.Taggable.
```

Comment:
Is there a reason `remote.Descriptor` doesn't just have these methods and satisfy this interface already?

Reply:
Name collision with the fields in v1.Descriptor that gets embedded... I have a lot of regrets.

Reply:
How do you feel about this? https://github.com/google/go-containerregistry/compare/master...jonjohnsonjr:slim-taggable